src/kernel/clock.c

```diff
@@ -39,6 +39,8 @@ closure_function(1, 1, boolean, timer_adjust_handler,
                 void *, v)
 {
     timer t = v;
+    if (t->absolute)
```

Comment:
Shouldn't this be ```if (!t->absolute)```?

Reply:
For a real-time-based timer, the expiry timestamp as calculated by timer_expiry() is (t->expiry - __vdso_dat->rtc_offset). When the system time is changed, __vdso_dat->rtc_offset is changed, so if t->expiry is adjusted by the same amount (and this is done for relative timers) the resulting expiry timestamp remains the same, whereas if t->expiry is not touched (for absolute timers) the expiry timestamp is affected by the change in the system time.